nsxt/data_source_nsxt_vpc_service_profile.go

```diff
@@ -18,14 +22,37 @@ func dataSourceNsxtVpcServiceProfile() *schema.Resource {
 			"description":  getDataSourceDescriptionSchema(),
 			"path":         getPathSchema(),
 			"context":      getContextSchemaExtended(true, false, false, true),
+			"is_default": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
 		},
 	}
 }
 
 func dataSourceNsxtVpcServiceProfileRead(d *schema.ResourceData, m interface{}) error {
-	_, err := policyDataSourceResourceRead(d, getPolicyConnector(m), getSessionContext(d, m), "VpcServiceProfile", nil)
+	// Using deprecated API because GetOk is not behaving as expected when is_default = "false".
+	// It does not return true for a key that's explicitly set to false.
+	value, defaultOK := d.GetOkExists("is_default")
+	_, dpOk := d.GetOk("display_name")
+	_, idOk := d.GetOk("id")
+
+	if defaultOK && !dpOk && !idOk {
+		query := make(map[string]string)
+		query["is_default"] = strconv.FormatBool(value.(bool))
+		_, err := policyDataSourceReadWithCustomField(d, getPolicyConnector(m), getSessionContext(d, m), "VpcServiceProfile", query)
+		return err
+	}
+	obj, err := policyDataSourceResourceRead(d, getPolicyConnector(m), getSessionContext(d, m), "VpcServiceProfile", nil)
 	if err != nil {
 		return err
 	}
+	converter := bindings.NewTypeConverter()
+	dataValue, errors := converter.ConvertToGolang(obj, model.VpcServiceProfileBindingType())
+	if len(errors) > 0 {
+		return errors[0]
+	}
+	vpcSvcProfile := dataValue.(model.TransitGateway)
```

Comment:
This should be a `model.VpcServiceProfile`, I guess that this is a copy-paste error